src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/BeneficialOwnerChangeService.java

```diff
@@ -0,0 +1,435 @@
+        .stream()
+        .map(this::covertBeneficialOwnerIndividualToChange)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Identifies and returns changes related to the 'Other' type beneficial owners in the
+   * submission.
+   *
+   * <p>This method is responsible for tracking the changes made to the beneficial owners of
+   * 'Other' type within the provided submission. Each change is encapsulated within a
+   * {@code BeneficialOwnerChange<OtherBeneficialOwnerPsc>} object.</p>
+   *
+   * @return A list of changes ({@code BeneficialOwnerChange<OtherBeneficialOwnerPsc>} objects)
+   * related to the beneficial owners of 'Other' type. If no changes are detected, an empty list is
+   * returned.
+   */
+  private List<Change> beneficialOwnerOtherChange() {
+    var beneficialOwnersGovernmentOrPublicAuthority = overseasEntitySubmissionDto
+        .getBeneficialOwnersGovernmentOrPublicAuthority();
+    return beneficialOwnersGovernmentOrPublicAuthority
+        .stream()
+        .map(this::covertBeneficialOwnerOtherChange)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Identifies and returns changes related to the 'Corporate' type beneficial owners in the
+   * submission.
+   *
+   * <p>This method is responsible for tracking the changes made to the beneficial owners of
+   * 'Corporate' type within the provided submission. Each change is encapsulated within a
+   * {@code BeneficialOwnerChange<CorporateBeneficialOwnerPsc>} object.</p>
+   *
+   * @return A list of changes ({@code BeneficialOwnerChange<CorporateBeneficialOwnerPsc>} objects)
+   * related to the beneficial owners of 'Corporate' type. If no changes are detected, an empty list
+   * is returned.
+   */
+  private List<Change> beneficialOwnerCorporateChange() {
+    var beneficialOwnersGovernmentOrPublicAuthority = overseasEntitySubmissionDto
+        .getBeneficialOwnersCorporate();
+    return beneficialOwnersGovernmentOrPublicAuthority
+        .stream()
+        .map(this::covertBeneficialOwnerCorporateChange)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Converts a BeneficialOwnerCorporateDto into a BeneficialOwnerChange object.
+   *
+   * <p>The method detects changes between the provided BeneficialOwnerCorporateDto and its
+   * corresponding information in the public and private data.</p>
+   *
+   * @param beneficialOwnerCorporateDto The BeneficialOwnerCorporateDto object to be converted.
+   * @return A BeneficialOwnerChange object if changes are detected, otherwise null.
+   */
+  public BeneficialOwnerChange<CorporateBeneficialOwnerPsc> covertBeneficialOwnerCorporateChange(
+      BeneficialOwnerCorporateDto beneficialOwnerCorporateDto) {
+
+    var beneficialOwnerChange = new CorporateBeneficialOwnerChange();
+
+    var psc = new CorporateBeneficialOwnerPsc();
+    beneficialOwnerChange.setPsc(psc);
```

Comment:
Should we be doing this later on if we've not updated the psc object yet?

Reply:
This shouldn't matter either way 
Java is pass-by-reference so as long as it's set you can still modify it and the psc in beneficialOwnerChange will be modified too 

If you think it's more readable to put it after everything is set I can but I thought it might be easier to set it as soon as it's created

Reply:
I think just for readability, would you mind moving this further down to when the object has been populated?

Reply:
Of course :)